jenkins/swift-functional-tests/50-run-scality-func-tests.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash -xue
+
+function ln_object_ring {
+    for i in {2..6}; do
+	local obj_link=/etc/swift/object-${i}.ring.gz
+	if ! [[ -e $obj_link ]]; then
+	    ln -s /etc/swift/object.ring.gz $obj_link
+	fi
+    done
+}
+
+# FIXME : this is already defined 10-install-ring.sh
+function get_AllowEncodedSlashes {
+    if is_ubuntu; then
+	echo "$UBUNTU_AllowEncodedSlashes"
+    elif is_centos; then
+	echo "$CENTOS_AllowEncodedSlashes"
+    else
+	echo "Unkown distribution"
+	return 1
+    fi
+}
+
+function initialize_env {
+    set +u
+    source devstack/openrc demo
+    OS_DEMO_USERNAME=$OS_USERNAME
+    OS_DEMO_PASSWORD=$OS_PASSWORD
+    OS_DEMO_TENANT_NAME=$OS_TENANT_NAME
+    source devstack/openrc admin
+    set -u
+    OS_ADMIN_USERNAME=$OS_USERNAME
+    OS_ADMIN_PASSWORD=$OS_PASSWORD
+    OS_ADMIN_TENANT_NAME=$OS_TENANT_NAME
+
+    source jenkins/openstack-ci-scripts/jenkins/distro-utils.sh
+    AllowEncodedSlashes=$(get_AllowEncodedSlashes)
+    # FIXME : this is already defined 10-install-ring.sh
+    HOST_IP=$(/sbin/ip addr show dev eth0 | sed -nr 's/.*inet ([0-9.]+).*/\1/p');
+    SPROXYD_NUMBER=2
+}
+
+function grant_permissions {
+    local user=$(whoami)
+    local directory
+    if  [[ -d /etc/httpd/conf.d ]]; then
+	directory="/etc/httpd/conf.d"
+    elif  [[ -d /etc/apache2/sites-enabled ]]; then
+	directory=/etc/apache2/sites-enabled
+    else
+	echo "Unkown distribution"
+	return 1
+    fi
+
+    sudo chown ${user} $directory
+    sudo chown ${user} /etc
```

Comment:
This is quite unsafe. If this ever gets executed on something else than a test VM, it will lead to weird things.

Could we instead use sudo each time we want to create something in /etc ?